airbyte-cdk/python/airbyte_cdk/sources/streams/utils/stream_helper.py

```diff
@@ -29,12 +29,26 @@ def get_first_stream_slice(stream) -> Optional[Mapping[str, Any]]:
 def get_first_record_for_slice(stream: Stream, stream_slice: Optional[Mapping[str, Any]]) -> StreamData:
     """
     Gets the first record for a stream_slice of a stream.
-    :param stream: stream
-    :param stream_slice: stream_slice
+
+    :param stream: stream instance from which to read records
+    :param stream_slice: stream_slice parameters for slicing the stream
     :raises StopIteration: if there is no first record to return (the read_records generator is empty)
     :return: StreamData containing the first record in the slice
     """
-    # We wrap the return output of read_records() because some implementations return types that are iterable,
-    # but not iterators such as lists or tuples
-    records_for_slice = iter(stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice))
-    return next(records_for_slice)
+    # Store the original value of exit_on_rate_limit
+    exit_on_rate_limit_has_setter = hasattr(stream.exit_on_rate_limit, "setter")
+
+    try:
+        # Ensure exit_on_rate_limit is safely set to True if possible
+        if exit_on_rate_limit_has_setter:
+            stream.exit_on_rate_limit = True  # type: ignore[misc]
+
+        # We wrap the return output of read_records() because some implementations return types that are iterable,
+        # but not iterators such as lists or tuples
+        records_for_slice = iter(stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice))
+
+        return next(records_for_slice)
+    finally:
+        # Restore the original exit_on_rate_limit value
+        if exit_on_rate_limit_has_setter:
+            stream.exit_on_rate_limit = False  # type: ignore[misc]
```

Comment:
What makes us believe the original value was `False`? Can't a connector be configured as `stream.exit_on_rate_limit = True` before calling this method?

Reply:
I see 
updated